sass/components/_chessboard.scss

```diff
@@ -0,0 +1,24 @@
+.chessboard {
+  height: 100vw;
+  width: 100vw;
+  display: grid;
+  grid-gap: 0;
+  grid-template-columns: repeat(8, auto);
+}
+
+.squares {
+  height: 12.5%;
+  width: 12.5%;
+
+  &--light {
+    background-image: url('../../static/img/board/light.png');
+    background-repeat: no-repeat;
+    background-size: cover;
+  }
+
+  &--dark {
+    background-image: url('../../static/img/board/dark.png');
+    background-repeat: no-repeat;
+    background-size: cover;
+  }
```

Comment:
Pola nie powinny mieć nazw klas tak nazwanych (jasne, ciemne). Co jeśli jedno pole będziemy chcieli zrobić pomarańczowe, a inne niebieskie? Raczej powinny być inne nazwy.

### Bardzo ważne pytanie. Czy kod JS musi wiedzieć o kolorach pól?
Raczej nie. Wręcz niewskazane. Po co mu zaprzątać głowę?
Dlatego kolory powinniśmy nadać w samym CSS bez dawania polom klas!

W linku wysyłam jak to zrobić.
https://www.w3schools.com/cssref/sel_nth-child.asp

Opis
za przykładem:
```
p:nth-child(odd) {
  background: red;
}

p:nth-child(even) {
  background: blue;
}
```

Reply:
Nie jestem mistrzem CSSa jak niektórzy więc się nie będę tutaj dużo wypowiadał :) 
Jak dla mnie to light / dark jest spoko np. nie od kolorów.
Ale to co Piotrek podesłał bardzo ciekawe - automatyzuje nam ta kwestie szachownicy :)

Reply:
Co do samych nazw, to nawet, jakby miały być niebieskie i pomarańczowe, to któreś musimy określić jako jasne/ciemne (hetman zawsze zaczyna na polu 'swojego' koloru). Specjalnie nie nazywałam białe/czarne. 
Co do samego przerzucenia tego do CSSa, to spoko, tylko (poprawcie mnie jeśli się mylę) ostatnio był chyba problem z przetestowaniem tła. Wiem, że sprawdzanie w testach tych klas najlepsze nie jest, ale zawsze to coś, a tak, bez testów manualnych, możemy te kolory pomieszać, co nie będzie dobre z punktu biznesowego.

Reply:
Tak, problem jest przetestować CSS - nie znam na to metody na razie :/ Wtedy chyba musimy zostawić bez testu - choć nie wierzę, że to mówię - no ale wygrałaś :) Front jest czesto problemem do przetestowania, tutaj można być mniej restrykcyjnym niż w logice.

Reply:
No ok, to zmienię, choć dla mnie - jako szachistki - kolory pól są dość istotne i powinno to być jakkolwiek przetestowane.

Reply:
Jak dla mnie - czy to zrobisz w JS czy w CSS bedzie dobrze - jesli przeknasz reszte zespolu :) 
Ja wlasnie przekonuje reszte do nazwy tabelki w bazie danych - i to nie jest proste :P

Reply:
Tzn. tak - jak dla mnie problem z testowaniem jest taki, że po prostu document, do którego dodajesz w tescie szachownice nie ma dodanego stylesheet. Co nie :) ? Bo jest 100% pusty. No i problem tutaj jest taki, że mamy SCSS, który w buildzie się zmienia do CSSa. I się zastanawiam jakby to mogło działać (na razie nie wiem) - żeby jednak ten SCSS się zrobił do CSSa i dodał w teście.

Reply:
@lamparina  Do nazwania jasny/ciemny zgadzam się. Masz rację!

Csss nie testujesz. Nie musisz. Testowanie kodu sprawdzane jest tylko w .JS i .TS

Jesli chodzi ustalenie klas (zamiast pseudoklas) to tak jak pisał Mateusz - "przekonaj mnie"! ;)

Reply:
Moim zdaniem korzystanie z nth-child, może być bardziej ryzykowne niż dodanie tych styli w JS, bo musisz najpierw ustalić w jakiej kolejności są tworzone pola szachownicy (tutaj akurat jadąc od ostatnich linii, tak jak normalny rzut z góry grając białymi - a8, b8, c8 ,(...) a1, b1, c1 itd.), potem ogarnąć, które pola (odd/even) powinny być jakiego koloru i dobrze je przypisać. Ja wiem, że to nie jest jakieś rocket science, ale potem, aby upewnić się, że pola są dobrze pokolorowane, musisz zbudować projekt i przekonać się **naocznie** (nie każdy gra w szachy, więc jeszcze powinien porównać, jak to powinno wyglądać, bo może nie wiedzieć). Dodając style w JS, można to zweryfikować w testach i gdyby ktoś coś spsuł, to dowiemy się o tym od razu i nie trzeba porównywać tego z przykładami z neta, bo oczekiwany kolor, jaki powinno mieć dane pole, jest już podany.

Żeby nie było, po części zgadzam się z Tobą, że normalnie to powinna być odpowiedzialność CSSa, natomiast w tym konkretnym przypadku, uważam, że ewentualny niewykryty błąd będzie poważniejszy - jako użytkownik może nie zwrócę uwagi, czy np. tło jest żółte zamiast brązowego (może tak miało być), ale jak zobaczę błędne rozstawienie na szachownicy (biały hetman na ciemnym polu), to potraktuję to jako braki merytoryczne.

Reply:
Przekonała mnie długość Twojego komentarza. ;)
Napraw jeszcze wyświetlanie się szachownicy i będzie git

Reply:
Już się wyświetla :) Przyśniło mi się rozwiązanie ;)

Reply:
Super, jeszcze (przynajmniej) jedna rzecz
`@media (orientation: landscape) {
   .szachownica {
       height: 100vh;
       width: 100vh;  
   }
}`
To dodaj, bo gdy ekran jest 'landscape' to nie widać całej szachownicy i trzeba przewijać

Reply:
Wiem, zauważyłam i poprawiłam :)

Reply:
2) Jak patrzę na szachownicę to widzę, że przydałoby się wrzucić tam zmienną która równałaby się jednemu bokowi szachownicy i na tej podstawie generować wielkość pól.
Zmienna może równać się przykładowo 100% szerokości kontenera lub 100% wysokości kontenera dla landscape. Będzie można to potem łatwo dostosować
- To taki must have

Reply:
Tylko chyba nie można wychwycić do zmiennej wartości rodzica do zmiennej. Dlatego też:
zamień może height i wight na  max-height, max, wight
I bedzie gites :)
Wtedy bedzie mieściło się zawsze w oknie, a i bedzie można z zewnątrz zmienić wymiary szachownicy.

No i oczywiscie nie zaomnij o @media

Reply:
Nie dziękuj :)